### ISSUE TITLE
Distributed Training Using Horovod  

### ISSUE THREAD

**Issue by jiangzihanict** [tensorpack/tensorpack#1238]
Hello, @ppwwyyxx sorry to bother you again.
We are going to train FasterRCNN on 4 nodes and each node has 8 GPUs. We bind one rank to one GPU. We have divided the whole dataset into 4 splits.  We found that the 8 ranks in a node read a copy of the corresponding split and the program seems stuck at the beginning of training. Do we need to divide the dataset according the rank.size() in order to ensure that each rank processes different data? How can we map the rankID to the corresponding data split？

**Comment by ppwwyyxx** [comment]
You can know your rank by `hvd.rank()`. It's up to you to decide whether you want to divide the dataset according to rank.
I cannot answer whether your code stucks without any details, unless you can fill out the issue template.

**Comment by jiangzihanict** [comment]
We encounter OOM error during training. Is the validation set too big? Our validation set is an (1460,1024,768,16) array. Our training image is (1024*768*16), batch_size = 1 and GPU memory is 12GB.

**Comment by ppwwyyxx** [comment]
I cannot diagnose anything from this limited information. My best guess is that your validation set is not large enough to cause OOM.
